bin/check-running-job

```diff
@@ -1,33 +1,55 @@
 #!/bin/bash
 
 function check_running_job {
+    [[ $# -eq 0 ]] && echo "Error: no app provided" && exit 1
+
+    # Get wait arugment if supplied
+    if [[ -n $2 && $2 == '--wait' ]]; then
+        wait_for_app='true'
+    fi
+
     # Input any app name
-    app_to_check=$1
+    if [[ -n "$1" ]]; then
+        if [[ $1 == '--wait' ]]; then
+            echo "Error: no app provided"
+            exit 1
+        fi
+        app_to_check=$1
+    fi
 
-    # Install jq
-    apt install -y jq
+    # Check jq
+    if ! command -v jq &> /dev/null
+    then
+        apt install -y jq
+    fi
 
     # Check if deployment is happening, if so ignore
     guid=$(cf app "$app_to_check" --guid)
     running=$(\
         cf curl "/v3/deployments?status_values=ACTIVE" \
-        | jq --arg guid "$guid" ".resources[] |
+        | jq --arg guid "$guid" \
+            ".resources[] |
             select(.relationships.app.data.guid == \$guid ) |
             .relationships.app.data.guid" \
     )
-    if [ "$running" != "" ]; then
+    if [[ $running != "" ]]; then
+        if [[ $wait_for_app == 'true' ]]; then
+                echo "Waiting on currently running job $app_to_check.."
+                sleep 10
+                check_running_job "$1" "$2"
+        else
             echo "Deployment in progress for $app_to_check, not doing anything"
-            exit 0
+            exit 1
+        fi
+    else
+        echo "No job running for app $app_to_check"
     fi
 
     # If there is more than one instance of 'instances', the app is already restarting
     if [[ $(cf app "$app_to_check" | grep -c '^instances:') -gt 1 ]]; then
       echo "Deployment or Restart in progress... not doing anything"
-      exit 0
+      exit 1
```

Comment:
Why are we going to exit 1 and raise an alarm?

Reply:
Hmm fair point. I guess if it is a normal restart we don't want to. 

> Pending the comments above, I'm okay with these changes. I just want to highlight that we can't use this for `catalog-fetch` and `catalog-gather` restarts. But those are technically broken and we have another ticket to address those
> 
> * #4187

Yup, noted [here](https://github.com/GSA/catalog.data.gov/pull/737/files#r1094896273). Will need to add there once addressed.